### DISCUSSION TITLE
filter by regex

### DISCUSSION THREAD

**Discussion by kineticz** [TomWright/dasel#177] [Q&A]
Suppose I have a json like this:

```json
{
  "users": [
    {
      "primary": true,
      "name": {
        "first": "Tom",
        "last": "Werknemer"
      }
    },
    {
      "primary": true,
      "name": {
        "first": "Tom",
        "last": "Wright"
      }
    },
    {
      "primary": false,
      "extra": {
        "name": {
          "first": "Joe",
          "last": "Blogs"
        }
      },
      "name": {
        "first": "Jim",
        "last": "Wright"
      }
    }
  ]
}
```

And I want to filter the users by a regex such as `/^W/`. Is this possible?

**Comment by TomWright** [comment]
This isn't possible yet - This is a potential feature though

**Reply by salim-b** [reply]
Corresponding issue with this feature request: https://github.com/TomWright/dasel/issues/183

**Reply by TomWright** [reply]
Not at the moment, but in v3 (which I'm actively developing) it will supported from the start

**Comment by TomWright** [comment]
Just a small update - I appreciate this has been a long time coming but with V3 (in active development) this is already supported:

Where `test.json` contains the JSON above....
```
$ cat test.json | ./dasel -i json 'users.filter(name.last =~ r/^W/)'
[{"name":{"first":"Tom","last":"Werknemer"},"primary":true},{"name":{"first":"Tom","last":"Wright"},"primary":true},{"extra":{"name":{"first":"Joe","last":"Blogs"}},"name":{"first":"Jim","last":"Wright"},"primary":false}]
```

V3 is being developed on this branch: https://github.com/TomWright/dasel/pull/432
